### PR TITLE
refactor: replace `.forEach()` with `for-of`

### PR DESCRIPTION
**Pull request by miniak**
Identified by https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md

Notes: none

### CODE REVIEW

**Comment by ckerr** [comment]
`darwin-testing-x64-tests` failure looks like a flake unrelated to this PR, but it's kind of interesting anyway:

```
Received signal 11 SEGV_MAPERR 1599183d4ea0
0   Electron Framework                  0x000000012fc6adb2 base::debug::CollectStackTrace(void**, unsigned long) + 18
1   Electron Framework                  0x000000012fc58a83 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x000000012fc6ad01 base::debug::(anonymous namespace)::StackDumpSignalHandler(int, __siginfo*, void*) + 1345
3   libsystem_platform.dylib            0x00007ff805d1cdfd _sigtramp + 29
4   ???                                 0x0000000000000030 0x0 + 48
5   Electron Framework                  0x0000000129db0bae electron::api::WebContents::New(v8::Isolate*, gin_helper::Dictionary const&) + 46
6   Electron Framework                  0x0000000129db0ef1 electron::api::WebContents::CreateFromWebPreferences(v8::Isolate*, gin_helper::Dictionary const&) + 337
7   Electron Framework                  0x0000000129dcbe76 electron::api::WebContentsView::New(gin_helper::Arguments*, gin_helper::Dictionary const&) + 22
8   Electron Framework                  0x0000000129dcc293 base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::Arguments*, gin_helper::Dictionary const&)>::Run(gin_helper::Arguments*, gin_helper::Dictionary const&) const & + 83
9   Electron Framework                  0x0000000129dcc121 void gin_helper::internal::InvokeNew<gin_helper::WrappableBase* (gin_helper::Arguments*, gin_helper::Dictionary const&)>(base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::Arguments*, gin_helper::Dictionary const&)> const&, v8::Isolate*, gin_helper::Arguments*) + 129
10  Electron Framework                  0x0000000129cec183 base::RepeatingCallback<void (v8::Isolate*, gin_helper::Arguments*)>::Run(v8::Isolate*, gin_helper::Arguments*) const & + 83
11  Electron Framework                  0x0000000129cec053 gin_helper::Dispatcher<void (v8::Isolate*, gin_helper::Arguments*)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 307
12  Electron Framework                  0x000000012bc7d0fa v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 266
13  Electron Framework                  0x000000012bc7b4dc v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, unsigned long*, int) + 1116
14  Electron Framework                  0x000000012bc7a3c6 v8::internal::Builtins::InvokeApiFunction(v8::internal::Isolate*, bool, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, int, v8::in
```

Possibly a nullptr dereference in `electron::api::WebContents::New()` ?

**Comment by ckerr** [comment]
`darwin-testing-x64-tests` CI flake in https://app.circleci.com/pipelines/github/electron/electron/73914/workflows/1a83002a-454e-404d-8865-82401eea4a48/jobs/1593874 appears to be the same as #39565

**Review by dsanders11 — APPROVED** [review]

File: docs/fiddles/features/web-hid/renderer.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
 async function testIt () {
   const grantedDevices = await navigator.hid.getDevices()
   let grantedDeviceList = ''
-  grantedDevices.forEach(device => {
+  for (const device of grantedDevices) {
     grantedDeviceList += `<hr>${device.productName}</hr>`
-  })
+  }
```
Comment (CaedenPH) [inline docs/fiddles/features/web-hid/renderer.js:6]: What about using `map` here?

**Review by CaedenPH — COMMENTED** [review]
Not sure if this is better, but what about using `map` instead of a `for-of` that performs an operation on every item

**Review by zcbenz — APPROVED** [review]

**Pull request merged**